From: Matthias Meier <Me...@FuH-E.de>
Date: Tue, 3 May 2011 08:49:45 +0200
Subject: Dynamic set ignores background-color on tbody-nodes

Fix the problem that tkhtml seems to ignore 'background-color' on tbody-tags.

http://groups.google.com/group/tkhtml3/browse_thread/thread/74ccb54074bec7d5
---
 src/htmlprop.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/htmlprop.c b/src/htmlprop.c
index a660990..aea01f5 100644
--- a/src/htmlprop.c
+++ b/src/htmlprop.c
@@ -2045,6 +2045,12 @@ HtmlComputedValuesInit(pTree, pNode, pParent, p)
         memcpy(&p->fontKey, pValues->fFont->pKey, sizeof(HtmlFontKey));
         pValues->mask = 
             (pValues->mask & iCopyMask) | (pParentValues->mask & !iCopyMask);
+        /* inherit background from parent */
+        p->values.cBackgroundColor->nRef++;
+        decrementColorRef(pTree, p->values.cBackgroundColor );
+
+        p->values.cBackgroundColor = 
+	    ((HtmlElementNode *)pParent)->pPropertyValues->cBackgroundColor;
     }
 
     p->values.cColor->nRef++;
